packages/next/src/server/after/builtin-request-context.ts

```diff
@@ -29,8 +30,20 @@ export type BuiltinRequestContext = {
   get(): BuiltinRequestContextValue | undefined
 }
 
+export type RunnableBuiltinRequestContext = BuiltinRequestContext & {
+  run<T>(value: BuiltinRequestContextValue, callback: () => T): T
+}
+
 export type BuiltinRequestContextValue = {
   waitUntil?: WaitUntil
   NextCacheHandler?: typeof CacheHandler
 }
 export type WaitUntil = (promise: Promise<any>) => void
+
+export function createLocalRequestContext(): RunnableBuiltinRequestContext {
+  const storage = createAsyncLocalStorage<BuiltinRequestContextValue>()
+  return {
+    get: () => storage.getStore(),
+    run: (value, callback) => storage.run(value, callback),
+  }
+}
```

Comment:
Is there a reason this is created here instead of what we've used in the past? Seems like we should use the existing patterns as much as possible.

Reply:
`globalThis[Symbol.for("@next/request-context")]` has a different signature from Node ALS, it uses `get` instead of `getStore()`, so this is just an adapter